lib/helper/Playwright.js

```diff
@@ -2655,6 +2661,265 @@ class Playwright extends Helper {
   async stopMockingRoute(url, handler) {
     return this.browserContext.unroute(...arguments);
   }
+
+  /**
+   * Starts recording of network traffic.
+   * This also resets recorded network requests.
+   *
+   * @return {Promise<void>}
+   */
+  async startRecordingTraffic() {
+    this._flushTraffic();
+    this.recording = true;
+    this.recordedAtLeastOnce = true;
+
+    this.page.on('requestfinished', async (request) => {
+      const information = {
+        url: request.url(),
+        method: request.method(),
+        requestHeaders: request.headers(),
+        requestPostData: request.postData(),
+      };
+
+      this.debugSection(`REQUEST: ${JSON.stringify(information)}`);
+
+      information.requestPostData = JSON.parse(information.requestPostData);
+      this.requests.push(information);
+      return this._waitForAction();
+    });
+  }
+
+  /**
+  *  Grab the recording network traffics
+  *
+  * @return { Promise<Array<any>> }
+  *
+  */
+  async grabRecordingTraffic() {
+    if (!this.recordedAtLeastOnce) {
+      throw new Error('Failure in test automation. You use "I.grabRecordingTraffic", but "I.startRecordingTraffic" was never called before.');
+    }
+    return this.requests;
+  }
+
+  /**
+   * Blocks traffic for URL.
+   *
+   * @param url URL to block . URL can contain * for wildcards. Example: https://www.example.com* to block all traffic for that domain.
+   */
+  async blockTraffic(url) {
+    this.page.route(url, (route) => {
+      route
+        .abort()
+        // Sometimes it happens that browser has been closed in the meantime. It is ok to ignore error then.
+        .catch((e) => {});
+    });
+    return this._waitForAction();
+  }
+
+  /**
+   * Mocks traffic for URL(s).
+   * This is a powerful feature to manipulate network traffic. Can be used e.g. to stabilize your tests.
+   *
+   * @param urls string|Array These are the URL(s) to mock, e.g. "/fooapi/*" or "['/fooapi_1/*', '/barapi_2/*']"
+   * @param responseString string The string to return in fake response.
+   * @param contentType Content type of response. If not specified default value 'application/json' is used.
+   */
+  async mockTraffic(urls, responseString, contentType = 'application/json') {
+    // Required to mock cross-domain requests
+    const headers = { 'access-control-allow-origin': '*' };
+
+    if (typeof urls === 'string') {
+      urls = [urls];
+    }
+
+    urls.forEach((url) => {
+      this.page.route(url, (route) => {
+        if (this.page.isClosed()) {
+          // Sometimes it happens that browser has been closed in the meantime.
+          // In this case we just don't fulfill to prevent error in test scenario.
+          return;
+        }
+        route.fulfill({
+          contentType,
+          headers,
+          body: responseString,
+        });
+      });
+    });
+    return this._waitForAction();
+  }
+
+  /**
+   * Resets all recorded network requests.
+   */
+  _flushTraffic() {
```

Comment:
I personally would make `I._flushTraffic` even officially public and remove the underscore. `_`.

I can imagine in some test scenarios users wants to be able to flush recorded traffic, for example if you pass a lot of pages during a long user journey, you might want to flush the traffic before entering next page to be able to assert similar traffic via "I.seeTraffic" on this 2nd page and not be fooled by some similar old traffic from 1st page which is still in array of recorded traffic.